cheroot/test/webtest.py

```diff
@@ -478,42 +477,38 @@ def shb(response):
     return resp_status_line, h, response.read()
 
 
-def openURL(
-    url, headers=None, method='GET', body=None,
-    host='127.0.0.1', port=8000, http_conn=http_client.HTTPConnection,
-    protocol='HTTP/1.1', raise_subcls=None, ssl_context=None,
-):
+def openURL(*args, raise_subcls=(), **kwargs):
     """
-    Open the given HTTP resource and return status, headers, and body.
+    Open a URL, retrying when it fails.
 
-    `raise_subcls` must be a tuple with the exceptions classes
-    or a single exception class that are not going to be considered
-    a socket.error regardless that they were are subclass of a
-    socket.error and therefore not considered for a connection retry.
+    Specify `raise_subcls` (class or tuple of classes) to exclude
+    those socket.error subclasses from being suppressed and retried.
     """
-    headers = cleanHeaders(headers, method, body, host, port)
-
-    # Trying 10 times is simply in case of socket errors.
-    # Normal case--it should run once.
-    for trial in range(10):
-        try:
-            return _open_url_once(
-                body, headers, host, http_conn, method,
-                port, protocol, ssl_context, url,
-            )
-        except socket.error as e:
-            if raise_subcls is not None and isinstance(e, raise_subcls):
-                raise
-            else:
-                time.sleep(0.5)
-                if trial == 9:
-                    raise
+    opener = functools.partial(_open_url_once, *args, **kwargs)
+
+    def on_exception():
```

Comment:
I'd rather have the exponential backoff implemented here...

Reply:
That's not unreasonable, but it's a feature request outside the scope of this change. This change will make it easier, however, to plug in different back off techniques. Good news is, someone has already written [BackoffDelay](https://tempora.readthedocs.io/en/latest/index.html#tempora.timing.BackoffDelay) for just this purpose.